File: api-php/samples/server/src/shared.php
<?php

use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;
use Trinsic\Api\Model\GetSessionResultRequest as GetSessionResultRequest;
use Trinsic\Api\Model\RecommendProvidersRequest as RecommendProvidersRequest;
use Trinsic\Api\Model\RecommendationInfo as RecommendationInfo;
use Trinsic\Api\Model\RecommendProvidersResponse as RecommendProvidersResponse;

return function ($app, $sessions) {
    

    $app->get('/redirect', function (Request $request, Response $response, $args) {
        $queryParams = $request->getQueryParams();
        
        $queryString = http_build_query($queryParams);
        
        $redirectUrl = '/redirect.html' . (!empty($queryString) ? '?' . $queryString : '');
        return $response
            ->withHeader('Location', $redirectUrl)
            ->withStatus(302);
    });

    $app->get("/providers", function (Request $request, Response $response, $args) use ($sessions) {
        $ipAddress = $request->getQueryParams()['ipAddress'] ?? null;

        $req = new RecommendProvidersRequest();
        $verificationProfileId = getenv('TRINSIC_VERIFICATION_PROFILE_ID');
        $req->setVerificationProfileId($verificationProfileId);

        if ($ipAddress) {
            $reqInfo = new RecommendationInfo();
            $reqInfo->setIpAddresses([$ipAddress]);
            $req->setRecommendationInfo($reqInfo);
        }

        $result = $sessions->recommendProviders($req);
    
        $response->getBody()->write(json_encode($result->jsonSerialize()));
        return $response->withHeader('Content-Type', 'application/json');
    });


    $app->post("/exchange-result", function (Request $request, Response $response, $args) use ($sessions) {
        $body = $request->getParsedBody();

        $exchangeRequest = new GetSessionResultRequest();
        $exchangeRequest->setResultsAccessKey($body['resultsAccessKey']);

        $result = $sessions->getSessionResult($body['sessionId'], $exchangeRequest);

        $response->getBody()->write(json_encode($result));
        return $response->withHeader('Content-Type', 'application/json');
    });


};

File: api-java/samples/server/trinsic/Shared.java
package id.trinsic;
import java.util.UUID;
import id.trinsic.api.SessionsApi;
import id.trinsic.api.models.GetSessionResultRequest;
import id.trinsic.api.models.RecommendationInfo;
import id.trinsic.api.models.RecommendProvidersRequest;
import id.trinsic.api.models.RecommendProvidersResponse;
import io.javalin.Javalin;
import java.util.UUID;
import java.util.List;

public class Shared{
    public static void SharedRoutes(Javalin app, SessionsApi session, UUID verificationProfileId){

        app.get("/providers", ctx -> {
            // Get query param (now optional)
            String ipAddress = ctx.queryParam("ipAddress");

            // Build request
            RecommendProvidersRequest req = new RecommendProvidersRequest();
            req.setVerificationProfileId(verificationProfileId);

            // Only set recommendation info if IP address is provided
            if (ipAddress != null && !ipAddress.isEmpty()) {
                RecommendationInfo info = new RecommendationInfo();
                info.setIpAddresses(List.of(ipAddress));
                req.setRecommendationInfo(info);
            }

            // Call the service
            RecommendProvidersResponse result = session.recommendProviders(req);

            // Return JSON response
            ctx.json(result);
        });

        app.post("/exchange-result", ctx -> {
            var body = ctx.bodyAsClass(id.trinsic.ExchangeResultRequest.class);

            var request = new GetSessionResultRequest();
            request.setResultsAccessKey(body.resultsAccessKey);

            var result = session.getSessionResult(UUID.fromString(body.sessionId), request);

            ctx.json(result);
        });
    }
}